src/components/Tree/Node.tsx

```diff
@@ -0,0 +1,91 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconProps, IconSize } from "@components/Icon/Icon";
+import { ReactComponent as IconCaretDown } from "@components/Icon/Svg/CaretDown.svg";
+import { ReactComponent as IconCaretRight } from "@components/Icon/Svg/CaretRight.svg";
+import { ReactElement, useEffect, useState } from "react";
+import css from "./Tree.module.css";
+
+export interface TreeNode {
+    id: string;
+    name: string;
+    icon?: ReactElement<IconProps>;
+    label?: string;
+    value?: string;
+    nodes?: TreeNode[];
+}
+
+interface NodeProps {
+    node: TreeNode;
+    strong?: boolean;
+    activeNodeId?: string;
+    onClick: (id: string) => void;
+}
+
+const getNodeClassNames = (active: boolean, strong: boolean, hasValue: boolean) => {
+    const classNames = [css.nodeLink];
+    if (active) {
+        classNames.push(css.nodeLinkSelected);
+    }
+    if (hasValue) {
+        classNames.push(css.selectableNodeLink);
+    }
+    if (strong) {
+        classNames.push(css.strong);
+    }
+
+    return classNames;
+};
+
+export default function Node({
```

Comment:
I wonder if it won't cause any conflict with [Node from the HTML API](https://developer.mozilla.org/docs/Web/API/Node) 😬

Reply:
to avoid any kind of conflict, i renamed it to TreeNode 6a5a786fe2e115deb012750be95160f80c31ad14  ;)